File: db4oj/core/src/com/db4o/internal/query/PredicateEvaluation.java
/* Copyright (C) 2004 - 2005  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.query;

import com.db4o.query.*;

/**
 * @exclude
 */
public class PredicateEvaluation implements Evaluation {
    
    public Predicate _predicate;
    
    public PredicateEvaluation(){
        // CS
    }
    
    public PredicateEvaluation(Predicate predicate){
        _predicate = predicate;
    }

    public void evaluate(Candidate candidate) {
        candidate.include(_predicate.appliesTo(candidate.getObject()));
    }

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Query/PredicateEvaluation.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Query;

namespace Db4objects.Db4o.Internal.Query
{
	/// <exclude></exclude>
	[System.Serializable]
	public class PredicateEvaluation : IEvaluation
	{
		public Predicate _predicate;

		public PredicateEvaluation()
		{
		}

		public PredicateEvaluation(Predicate predicate)
		{
			// CS
			_predicate = predicate;
		}

		public virtual void Evaluate(ICandidate candidate)
		{
			candidate.Include(_predicate.AppliesTo(candidate.GetObject()));
		}
	}
}
